File: src/main/java/net/jeikobu/jbase/config/IGlobalConfig.java
package net.jeikobu.jbase.config;

import java.util.Locale;

public interface IGlobalConfig {
    String getDefaultCommandPrefix();
    String getToken();
    boolean useDefaultCommands();
    Locale getGlobalLocale();
    <T> T getValue(String key, Class<T> valueType);
}

File: src/main/java/net/jeikobu/kotomi/base/config/IGlobalConfig.kt
package net.jeikobu.kotomi.base.config

import java.util.*

interface IGlobalConfig {
    val defaultCommandPrefix: String
    val token: String
    val useDefaultCommands: Boolean
    val globalLocale: Locale

    fun <T> getValue(key: String, valueType: Class<T>): T
}